AnySerial.h

```diff
@@ -17,30 +17,31 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- * 
+ *
  * Wrapper class around:
  * SoftwareSerial
  * AltSoftSerial
  * HardwareSerial (Serial, Serial[123])
  *
- * The goal of AnySerial is to make writing libraries which 
+ * The goal of AnySerial is to make writing libraries which
  * use the serial port more cross platform and easier to use.
  *
- * Rather then hard coding a specific Serial port or pins, 
- * you can now just accept an AnySerial object to your 
- * constructor and you don't care what what kind of serial 
+ * Rather then hard coding a specific Serial port or pins,
+ * you can now just accept an AnySerial object to your
+ * constructor and you don't care what what kind of serial
  * port it is (hardware or serial), you can just interact
  * with it normally.
  *
  * Just be sure to import the necessary SoftwareSerial.h or
- * AltSoftSerial.h BEFORE importing AnySerial.h so that 
+ * AltSoftSerial.h BEFORE importing AnySerial.h so that
  * AnySerial will be compiled with the necessary support.
  *
- * AnySerial always supports hardware serial ports (Serial, 
+ * AnySerial always supports hardware serial ports (Serial,
  * Serial1, etc)
  */
 
 #include <HardwareSerial.h>
+#include <SoftwareSerial.h>
```

Comment:
Read the comment above. You need to import SoftwareSerial.h BEFORE AnySerial.h because you may or may not require it and I didn't want to force it.

Reply:
I am `#import <SoftwareSerial.h>`ing **before** `#import <AnySerial.h>`ing, but it keeps saying `undefined reference to 'AnySerial::AnySerial(SoftwareSerial*)'` which AFAIK means the `#ifdef SoftwareSerial_h ... #endif` in your library don't work because the `SoftwareSerial_h`-macro is unset during compilation of your library for whatever reason. In the Arduino IDE 1.0.5 this macro is definitely set in the `SoftwareSerial` library.